src/render/md.jl

```diff
@@ -1,43 +1,30 @@
 ## Docs-specific rendering
 
-function writemime(io::IO, mime::MIME"text/md", docs::Docs{:md}, mdstyle::Dict{Symbol, ByteString} = DEFAULT_MDSTYLE)
+function writemime(io::IO, mime::MIME"text/md", docs::Docs{:md})
     println(io, docs.data)
 end
 
 ## General markdown rendering
 
-const DEFAULT_MDSTYLE = Dict{Symbol, ByteString}([
-  (:header         , "#"),
-  (:objname        , "####"),
-  (:meta           , "**"),
-  (:exported       , "##"),
-  (:internal       , "##"),
-])
-
 const HTAGS = ["#", "##", "###", "####", "#####", "######"]
 const STYLETAGS = ["", "*", "**"]
 
-print_help(io::IO, cv::ASCIIString, item) = cv in HTAGS ? println(io, "$cv $item") : println(io, cv, item, cv)
+print_help(io::IO, cv::ASCIIString, item) = cv in HTAGS              ?
+                                            println(io, "$cv $item") :
+                                            println(io, cv, item, cv)
 
-function validate(mdstyle::Dict{Symbol, ByteString})
-    reg_keys = keys(DEFAULT_MDSTYLE)
-    length(keys(mdstyle)) != length(reg_keys) && error(
-            "`mdstyle` expected number of keys: $(length(reg_keys))  Got: $(length(keys(mdstyle)))")
-    vaild_tags = vcat(HTAGS,STYLETAGS)
-    for (k, v) in mdstyle
-        k in reg_keys || error("Invalid mdstyle key:  `$k`. Valid keys: [$(join(reg_keys, ", "))].")
-        v in vaild_tags || error("Invalid mdstyle value:  `$v`. Valid values: [$(join(vaild_tags, ", "))].")
+function save(file::String, mime::MIME"text/md", doc::Metadata, config::Config)
+    # validate mdstyle
+    for k in [:mdstyle_header, :mdstyle_objname, :mdstyle_meta, :mdstyle_exported, :mdstyle_internal]
+       getfield(config, k) in vcat(HTAGS,STYLETAGS) ||
+                            error("""Invalid mdstyle value: config-item `$k -> $(config[k])`.
+                                    Valid values: [$(join(vcat(HTAGS,STYLETAGS), ", "))].""")
```

Comment:
I was thinking the same but thought maybe as it only related to markdown I move it there. Anyway I will change it as you suggested.